lib/src/cli.rs

```diff
@@ -879,6 +878,18 @@ where
     run_from_opt(Opt::parse_including_static(args)).await
 }
 
+/// Find the base binary name from argv0 (without a full path). The empty string
+/// is never returned; instead a fallback string is used.
+fn callname_from_argv0(argv0: &OsStr) -> &str {
+    let default = "bootc";
+    let argv0 = argv0.to_str().filter(|s| !s.is_empty()).unwrap_or(default);
```

Comment:
This is pretty semantically dense...

Reply:
My understanding of the logic here: if argv0 fails to convert to `str`, or in case that string is empty, use a default value, and then try to split it based on `/`, taking the right side, but if the split fails or the right side is empty, use the original (or default) string instead.

It gets difficult to read because of some non-obvious stuff like `.to_str()` returning an `Option<>` and `.filter()` working on `Option<>` as if it were a 0-or-1-item list.  To me it looks like you're trying to filter non-empty ... characters?  No wait, that doesn't make sense...

I feel like you could use a bit more early returning and/or open-coding of stuff.  Even a bit of `match` could help here... and isn't there some `basename` type feature somewhere we can use?

This is obviously entirely stylistic...

Reply:
This function will also return `/` for the input `/`, but I think maybe you'd prefer `bootc` in that case?  ie: if the string successfully splits but the right-hand result of the splitting is empty maybe you should prefer a fallback to the default instead of taking the entire string?

Reply:
> This is pretty semantically dense...

Heh for sure.

Reply:
I reworked this one to use `Path::file_name()` and yeah it got way cleaner.